### PR TITLE
Infra 7690 ssl

### PR DESCRIPTION
**Pull request by ebedarev**
- SSL support
- Swap producers on SIGHUP
- Kafka test proxy pkg to simulate connection drop and verify TLS auth
- Kafka pkg tests

### CODE REVIEW

File: pkg/kafka/kafka.go [hunk 1]
```diff
@@ -27,42 +28,170 @@ const (
...
+		p.producerWg.Done()
+	}(&pw)
+	p.producerMutex.Unlock()
+	activeProducer.With(prometheus.Labels{
+		"producer_id": strconv.FormatUint(uint64(pid), 10),
+	}).Set(1)
+	lastProducerStartTime.Set(float64(time.Now().Unix()))
+	if KafkaParamsPathExists(kafkaParams, "ssl.certificate.location") {
+		certET, err := ParamsCertExpirationTime(kafkaParams, "ssl.certificate.location")
+		if err == nil {
```
Comment (onorua) [inline pkg/kafka/kafka.go]: what will happen if there will be error?

File: pkg/kafka/kafka.go [hunk 2]
```diff
@@ -27,42 +28,170 @@ const (
...
+			metricCertExpirationTime.Set(float64(certET.Unix()))
+		}
+	}
+	if KafkaParamsPathExists(kafkaParams, "ssl.ca.location") {
+		caET, err := ParamsCertExpirationTime(kafkaParams, "ssl.ca.location")
+		if err == nil {
```
Comment (onorua) [inline pkg/kafka/kafka.go]: What will happen in case of error?

File: pkg/kafka/kafka.go [hunk 3]
```diff
@@ -27,42 +28,170 @@ const (
...
+				go func() {
+					time.Sleep(10 * time.Minute)
+					msgInTransit.Delete(prometheus.Labels{
+						"producer_id": strconv.FormatUint(uint64(oldPW.ID), 10),
+					})
+					activeProducer.Delete(prometheus.Labels{
+						"producer_id": strconv.FormatUint(uint64(oldPW.ID), 10),
+					})
+				}()
+				p.producerMutex.Lock()
```
Comment (onorua) [inline pkg/kafka/kafka.go]: not really sure why do you need Lock/Unlock without anything in between?

File: pkg/kafka/kafka.go (hunk 1 above)
Comment (ebedarev) [reply pkg/kafka/kafka.go]: This section is needed only to update the metric. If there's an error, metric stays unchanged. Maybe it makes sense to reset the metric if any error?

File: pkg/kafka/kafka.go (hunk 2 above)
Comment (ebedarev) [reply pkg/kafka/kafka.go]: Same stuff with the metric update

File: pkg/kafka/kafka.go (hunk 3 above)
Comment (ebedarev) [reply pkg/kafka/kafka.go]: fixed

File: pkg/kafka/kafka.go (hunk 1 above)
Comment (onorua) [reply pkg/kafka/kafka.go]: yes, reset could identify the problem if certs were already read, what will happen if it is first iteration? And the metric is anyway `nil`?
Maybe we could reset the metric and increase some metric, such as `cert_expiration_check_errors` or similar one?

**Pull request merged**